pulsar-client/src/main/java/org/apache/pulsar/client/impl/transaction/TransactionImpl.java

```diff
@@ -164,6 +168,7 @@ public synchronized void registerCumulativeAckConsumer(ConsumerImpl<?> consumer)
         return checkIfOpenOrCommitting().thenCompose((value) -> {
             CompletableFuture<Void> commitFuture = new CompletableFuture<>();
             this.state = State.COMMITTING;
+            timeout.cancel();
```

Comment:
Why cancel timeout task here, not after state is in `COMMITTED, ABORTED, ERROR` which this transaction is actually completed.

Reply:
The function of timeout is to set the status of the transaction to timeout when the transaction times out.
If the state of the transaction has become committing or aborting, the timeout will have no effect. At this point, the resources of this timetask can be released